src/core/email-client.js

```diff
@@ -53,101 +70,259 @@ function getFreshImap (cb) {
 }
 
 class EmailClient {
-    constructor () {
+    //constructor takes the full name of user as String
+    constructor (name) {
         var self = this;
-
-        //  Get Imap obj and check emails
-        getFreshImap(function (err, imap) {
+        self._name = name; 
+        self._email = "";
+        self._imap = Q.defer();
+        self._transporter = Q.defer();
+        self._status = Q.defer();
+       
+        getFreshImap(function (err, imap, email, smtpConfig) {
             if (err) {
                 console.log(err);
             } else {
-                self.imap = imap;
-                self._checkEmails();
+                self._imap.resolve(imap);
+                self._email = email;
+                self._transporter.resolve(NodeMailer.createTransport(smtpConfig));
+                self._ready = true;
+                //console.log('Completed');
+                
             }
         });
+
+    }
+
+    _loadInfo(){
+        //  Get Imap obj and check emails
+        var self = this;
+        return self._imap.promise;
+    }
+
+    _loadTransporter(){
+        var self = this; 
+        return self._transporter.promise;
+    }
+
+    _loadStatus(){
+        var self = this;
+        return self._status.promise;
+    }
+
+    //method to change name of user
+    _changeName(name){
+        var self = this;
+        self._name = name;
     }
 
-    _checkEmails () {
+    _changeEmail(email){
```

Comment:
Yea. I forgot about that. I will get new imap and other stuff is necessary to adopt with this new information